bgpd/bgp_route.c

```diff
@@ -2219,8 +2219,16 @@ bool subgroup_announce_check(struct bgp_dest *dest, struct bgp_path_info *pi,
 	 * implementations.
 	 */
 	if (CHECK_FLAG(bgp->flags, BGP_FLAG_EBGP_REQUIRES_POLICY))
-		if (!bgp_outbound_policy_exists(peer, filter))
+		if (!bgp_outbound_policy_exists(peer, filter)) {
+			if (monotime_since(&bgp->ebgprequirespolicywarning,
+					   NULL) > SIXHOUR2USEC ||
+			    bgp->ebgprequirespolicywarning.tv_sec == 0) {
+				zlog_warn(
+					"EBGP inbound/outbound policy not properly setup, please configure in order for your peering to work correctly");
```

Comment:
Sounds good to me :+1:, just maybe would be worth warning about the `outbound` direction here? And the same for `inbound` below?

Reply:
I didn't want to make it specific to one direction since it's one timer for both.  Was trying to avoid the situation where they fix one direction and then fix the other in 6 hours.  Especially since most of the time it's the operator not configuing any policy and nothing is working